src/CodeGen_Internal.cpp

```diff
@@ -474,5 +474,46 @@ void set_function_attributes_for_target(llvm::Function *fn, Target t) {
     fn->addFnAttr("reciprocal-estimates", "none");
 }
 
+Expr compile_unsafe_promises(const Call *op, bool check_unsafe_promises) {
+    Expr result;
+
+    if (op->is_intrinsic(Call::unsafe_promise)) {
+        if (check_unsafe_promises) {
+            std::ostringstream promise_expr_text;
+            promise_expr_text << op->args[0];
+            Expr cond_as_string = StringImm::make(promise_expr_text.str());
+            Expr promise_broken_error =
+                Internal::Call::make(Int(32),
+                                     "halide_error_requirement_failed",
```

Comment:
Do we (eventually) need a unique error message here, or is recycling requirement_failed ok? (I think it probably is, just noting for discussion)